design/design-doc-smaller-providers.md

```diff
@@ -0,0 +1,704 @@
+  `upbound/provider-aws`.
+* Share a `ProviderConfig` by taking a dependency on a (e.g.)
+  `provider-aws-config` package.
+* Continue to use contemporary “compiled-in” cross-resource-references, short to
+  medium term.
+
+Provider generation tooling like [`upjet`][upjet] would be updated to support
+generating service-scoped providers.
+
+No changes would be required to Crossplane core - i.e. to the package manager.
+No changes would be required to the long tail of existing, smaller Crossplane
+providers such as `provider-terraform` and `provider-kubernetes`.
+
+The following are the high level pros and cons of the proposed approach.
+
+Pros
+
+* No new concepts or configuration required.
+* Fewer CRDs is the default user experience
+* Simple, non-disruptive migration path from existing providers.
+* Folks can create their own “meta-providers” (e.g. `acmeco/provider-aws`) that
+  pull in just the services they want. This is just packaging - no compiling
+  required.
+* Provider work spread over many provider processes (i.e. pods) should scale
+  better.
+* Granular deployments. You could upgrade the EKS provider without having to
+  upgrade RDS. 
+* No major changes to Crossplane core, which means the rollout is uncoupled from
+  Crossplane's quarterly release cycle.
+
+Cons
+
+* Granular deployments. More providers to upgrade, monitor, etc.
+* Does not achieve a perfect 1:1 ratio of installed-to-used CRDs.
+* The average deployment may require more compute resources, short term.
+
+I'd like to expand on the first pro a little. No new concepts or configuration
+required. To me, this is the key benefit of the proposed approach, and the
+reason I believe it's the best option for our project and community, long term.
+
+Imagine you're learning how Crossplane works for the first time. The first two
+steps on your getting started checklist are probably:
+
+1. Install Crossplane.
+2. Install the Provider(s) for the things you want to manage.
+
+Keep in mind today that the things you might want to manage might be a cloud
+provider like AWS, but it might also be Helm charts, SQL databases, etc. Most
+folks today are running more than one provider.
+
+Under this proposal, this does not change. You'd explain how to get started with
+Crossplane in exactly the same way. The only difference is that in some cases
+you'd think about "the things you want to manage" in a more granular way. You'd
+think "I want RDS, ElastiCache, CloudSQL, GKE, and Helm" not "I want AWS, GCP,
+and Helm".
+
+I would argue that most folks already think this way. Folks don't think "I want
+AWS support" without having some idea which AWS _services_ they want to use.
+Confirming "Does it support AWS?" is a tiny speed bump before you confirm "Does
+it support RDS?" (or whatever services you need).
+
+### Community Impact
+
+Based on a survey of ~40 community members, the average deployment would:
+
+* Install ~100 provider CRDs, down from over 900 today.
```

Comment:
Is this number updated after the recently added [Large Services](https://github.com/crossplane/crossplane/pull/3939/files?diff=unified&w=0#diff-c35d366039b2ed54517b98b83bc01e4e421bae2ea9ed3d0ee53a9231afcaf8d3R193) section? If not, this would go further down.

Reply:
I think it was actually miscalculated before. 😞  I went to update it and found that _with_ the new Large Services plan the average was ~100. I will double-check my calculations today.